### PR TITLE
Docs site poc

### PR DESCRIPTION
**Pull request by andrewm4894**
related to this issue: https://github.com/LAION-AI/Open-Assistant/issues/340

you can see the docs site create by this branch here: https://andrewm4894.github.io/Open-Assistant/

changes:
- add docusaurus project under /docs
- add gh action to deploy changes under /docs
- run yarn install within devcontainer so users can just do `cd docs; yarn start` to work on docs and see impacts if they want.

to try this out

create codespace on https://github.com/andrewm4894/Open-Assistant/tree/docs-site-poc

```bash
cd docs
yarn install
yarn start
```

Then the docs site will by default be served on port 3000 so you can look in there and changes should be reflected automatically. 

So as you work on docs then you can see impact straight away. 

image

### CODE REVIEW

**Comment by andrewm4894** [comment]
@yk i think this is pretty much ready for review. 

I see a few PR's updating .md files etc so if we decide to get a docs site like this going i can handle any conflicts we might have initially. Just thinking be handy to put it in place sooner (if that's decided) as i can imagine will be lots of docs changes. 

Last thing i will do is update /docs/README.md to try make it clear and obvious how to make additions or changes to the docs.

**Comment by andrewm4894** [comment]
if we merge this then someone with access would just need to go to the github pages part of the repo settings and do this:

image

We would also need to make one last commit (or small follow on PR) - to update all the `andrewm4894` parts of the docusaurus config with `LAION-AI`. (just have to use andrewm4894 for now to make sure my example site builds from my fork).

File: docs/.gitignore [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Dependencies
+docs/node_modules
```
Comment (bitplane) [inline docs/.gitignore:2]: `node_modules` will ignore `docs/node_modules` and the same for the other dupes.

**Review by bitplane — COMMENTED** [review]
I don't really have the skill level to review js/ts files so just stuck my dirty oar in about .gitignore while reading about docusaurus. It looks really cool. Nice work :)

**Comment by andrewm4894** [comment]
@yk @andreaskoepf what you think about this - i'm happy to take on any donkey work around maintaining it, helping users add to it and things like that. Just thinking a decent docs site might help onboard people easier over next while and maybe help them to not have to dig through .md files on GH (which they could still do too if they wanted since this really just about packaging up existing .md files to given them a bit of an obvious structure).

File: .devcontainer/post_create_command.sh [hunk 2]
```diff
@@ -1,2 +1,7 @@
 # ensure pre-commit is installed
 pre-commit install
+
+# run yarn install in docs folder
+cd docs
+yarn install
+cd ..
```
Comment (yk) [inline .devcontainer/post_create_command.sh:7]: maybe we can leave this away? how much time does it add?

File: docs/docusaurus.config.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+// Note: type annotations allow type checking and IDEs autocompletion
+
+const lightCodeTheme = require("prism-react-renderer/themes/github");
+const darkCodeTheme = require("prism-react-renderer/themes/dracula");
+
+/** @type {import('@docusaurus/types').Config} */
+const config = {
+  title: "Open Assistant",
+  tagline: "Build the assistant of the future!",
+  url: "https://andrewm4894.github.io",
```
Comment (yk) [inline docs/docusaurus.config.js:11]: this probably needs to become the main url :)

**Review by yk — APPROVED** [review]
Very cool, thank you!

File: .devcontainer/post_create_command.sh (hunk 2 above)
Comment (andrewm4894) [reply .devcontainer/post_create_command.sh:7]: Takes a few seconds, maybe worth having in default container to make it super easy for people to build docs site in a codespace if they want. Of course all they read need do is change a .md file and the workflow will build or fail so if not like they need to. 

But maybe handy to have there to make docs contributions as frictionless as possible.

File: docs/docusaurus.config.js (hunk 3 above)
Comment (andrewm4894) [reply docs/docusaurus.config.js:11]: Yep, I can change all that in one last commit now before we merge this. Was just leaving like that so my site was working. 

Will make follow up commit to this branch to replace all the andrewm4894 with LAION-AI

**Comment by andrewm4894** [comment]
I will resolve conflicts and finalize the urls in one last commit. Will ping again once done.

**Pull request merged**